### PR TITLE
Transform expand provider to converter

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: service/internal/configprovider/expand.go [hunk 1]
```diff
@@ -15,45 +15,20 @@
...
-	"go.opentelemetry.io/collector/config/configmapprovider"
 )
 
-type expandMapProvider struct {
-	base configmapprovider.Provider
-}
-
-// NewExpand returns a Provider, that expands all environment variables for a
+// NewExpandConverter returns a Provider, that expands all environment variables for a
 // config.Map provided by the given Provider.
```
Comment (dmitryax) [inline service/internal/configprovider/expand.go:24]: Please update the comment

**Review by dmitryax — APPROVED** [review]

**Review by Aneurysm9 — DISMISSED** [review]
#4600 is going to break things for downstream distributions and we need to pause on digging a deeper hole.

**Comment by bogdandrutu** [comment]
@Aneurysm9 if still unclear, see the next PR #4637. Also as commented in the #4600 it is not as bad as you think, and with #4637 you will not miss anything.

**Comment by Aneurysm9** [comment]
Perhaps we should add this to the discussion on https://github.com/open-telemetry/opentelemetry-collector/issues/4630.  With all of the fragmented PRs happening in this space with zero communication of plan or desired end-state, I have no idea what works and what doesn't.  I will review this in the morning when I return to work.

**Comment by bogdandrutu** [comment]
@Aneurysm9 this is not a component FYI, so your argument does not stand for that issue.

**Comment by bogdandrutu** [comment]
@Aneurysm9 anything else blocking this or can be merged?

**Pull request merged**